src/board/moves.rs

```diff
@@ -23,11 +26,21 @@
 /// assert_eq!(chess_move.initial_square, 8);
 /// assert_eq!(chess_move.target_square, 24);
 /// ```
-#[derive(PartialEq, Debug)]
+#[derive(PartialEq, Debug, Eq, Hash, Clone)]
 pub struct Move {
     /// The index of the initial square where the move starts.
     pub initial_square: u8,
 
     /// The index of the target square where the move ends.
     pub target_square: u8,
 }
+
+// Needed to compare two list of moves with different order of moves
+// Allow dead code because test is not enough to realise it is being used
+#[allow(dead_code)]
+pub fn are_moves_equal(moves1: &[Move], moves2: &[Move]) -> bool {
```

Comment:
I would also consider comparing the sorted vectors of moves. 
But since this is only in test code I don't think it matters much